packages/core/src/utils/security/handlers/utils.ts

```diff
@@ -0,0 +1,39 @@
+import * as Either from 'fp-ts/lib/Either';
+import { AuthErr } from './types';
+
+const invalidCredsErr = Either.left({
+  name: 'Forbidden',
+  status: 403,
+  message: 'Invalid credentials used',
+  headers: {},
+});
+
+export function genRespForScheme<R>(isSchemeProper: boolean, isCredsGiven: boolean, resource: R, msg: string) {
+  const handler = [
+    {
+      test: () => isSchemeProper && isCredsGiven,
+      handle: () => Either.right(resource),
+    },
+    {
+      test: () => isSchemeProper,
+      handle: () => invalidCredsErr,
+    },
+  ].find(possibleHandler => possibleHandler.test());
+
+  return handler ? handler.handle() : Either.left(genUnauthorisedErr(msg));
+}
+
+export const genUnauthorisedErr = (msg: string): AuthErr => ({
+  name: 'Unauthorised',
+  message: 'Invalid security scheme used',
+  status: 401,
+  headers: msg ? { 'WWW-Authenticate': msg } : {},
+});
+
+export function isScheme(authScheme: string, shouldBeScheme: string) {
+  return (authScheme || '').toLowerCase() === shouldBeScheme;
+}
+
+export function when<R>(isOk: boolean, msg: string, resource?: R) {
```

Comment:
Is it really worth to make this a separate function? This function is a little bit confusing to me.

```suggestion
export function when<R>(condition: boolean, errorMessage: string, resource?: R) {
```

Reply:
It's used a couple of times and I find it quite useful in terms of tidying things up. But if you'd rather see it removed, LMK :)